components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/micro/integrator/inbound/endpoint/protocol/grpc/GRPCInjectHandler.java

```diff
@@ -0,0 +1,203 @@
+
+    /**
+     * This method will be called when the initiated gRPC call is not expecting an response.
+     * This will inject the gRPC message to the to the message sequence as well as the responseObserver .
+     * inject the message.
+     * @param receivedEvent received event from gRPC
+     * @param responseObserver object to be used to send message processing error back to gRPC client
+     */
+    public void invokeConsume(Event receivedEvent, StreamObserver<Empty> responseObserver) {
+        try {
+            initiateSequenceAndInjectPayload(responseObserver, receivedEvent, createMessageContext());
+        } catch (AxisFault e) {
+            log.error("Error while consuming the Grpc Message", e);
+            throw new SynapseException("Error while consuming the JMS Message", e);
+        }
+    }
+
+    private void initiateSequenceAndInjectPayload(StreamObserver responseObserver,
+                                                  Event receivedEvent,
+                                                  org.apache.synapse.MessageContext msgCtx) throws AxisFault {
+        String msgPayload = receivedEvent.getPayload();
+        String sequenceName = receivedEvent.getHeadersMap().get(HEADER_MAP_SEQUENCE_PARAMETER_NAME);
+        SequenceMediator seq;
+        if (sequenceName != null) {
+            if (log.isDebugEnabled()) {
+                log.debug(sequenceName + " sequence, received via gRPC headers.");
+            }
+            seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(sequenceName);
+        } else {
+            if (injectingSeq == null || injectingSeq.isEmpty()) {
+                log.error("Sequence name is not specified in inbound endpoint or empty.");
+                return;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug(injectingSeq + " sequence, received via the inbound endpoint.");
+            }
+            seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(injectingSeq);
+        }
+        msgCtx.setProperty(SynapseConstants.IS_INBOUND, true);
+        //validating the sequence
+        if (seq != null) {
+            if (!seq.isInitialized()) {
+                seq.init(synapseEnvironment);
+            }
+            seq.setErrorHandler(onErrorSeq);
+            if (log.isDebugEnabled()) {
+                log.debug("injecting received gRPC message to sequence : " + injectingSeq);
+            }
+            if (!synapseEnvironment.injectInbound(msgCtx, seq, this.sequential)) {
+                return;
+            }
+        } else {
+            log.error("Sequence: " + injectingSeq + " not found");
+        }
+        MessageContext axis2MsgCtx =
+                ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx).getAxis2MessageContext();
+        //setting transport headers
+        axis2MsgCtx.setProperty(MessageContext.TRANSPORT_HEADERS
+                , receivedEvent.getHeadersMap());
```

Comment:
Incorrect formatting. Let's fix

Reply:
fixed in https://github.com/wso2/micro-integrator/pull/854/commits/fff5493dd60f14989271869d8d399bb396a56033